syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";
import "google/protobuf/struct.proto";

message Azure {
  enum TaskType {
    UNKNOWN = 0;
    FILTER_LOG_EVENTS = 1;
    FETCH_METRICS = 2;
  }

  message FilterLogEvents {
    google.protobuf.StringValue workspace_id = 1;
    google.protobuf.StringValue filter_query = 2;
    google.protobuf.StringValue timespan = 3;
  }

  message FetchMetrics {
    google.protobuf.StringValue resource_id = 1;
    google.protobuf.StringValue metric_names = 2;
    google.protobuf.StringValue timespan = 3;
    google.protobuf.StringValue aggregation = 4;
    google.protobuf.StringValue granularity = 5;
  }

  TaskType type = 1;
  oneof task {
    FilterLogEvents filter_log_events = 2;
    FetchMetrics fetch_metrics = 3;
  }
}